### PR TITLE
Added upgraded and normal weapons

### PR DESCRIPTION
**Pull request by TubBuddy**
I separated upgraded and normal weapons along with Samantha box and Winters Howl. I tried to add the Matryoshka dolls but no luck i tried all sorts of combinations but i just couldn't make it work.

### CODE REVIEW

File: scripts/menu.gsc [hunk 1]
```diff
@@ -536,10 +536,39 @@ MenuOptionsPlayer(menu, player)
             self addMenu(menu, "Upgraded Specials");
             self addOpt("Zitros Orbital Arbalest", &GiveClientWeapon, "launcher_standard_t8_upgraded", self);
             self addOpt("Thekrauss Refibrillator++", &GiveClientWeapon, "special_ballisticknife_t8_dw_upgraded", self);
-            if(BO4GetMap() == "Blood" || BO4GetMap() == "AO" || BO4GetMap() == "Tag" || BO4GetMap() == "Classified"){
-                self addOpt("Porters X2 Ray Gun", &GiveClientWeapon, "ray_gun_upgraded", self);
-                self addOpt("Porters Mark II Ray Gun", &GiveClientWeapon, "ray_gun_mk2_upgraded", self);
+            self addOpt("Porters X2 Ray Gun", &GiveClientWeapon, "ray_gun_upgraded", self);
+            self addOpt("Porters Mark II Ray Gun", &GiveClientWeapon, "ray_gun_mk2_upgraded", self);
```
Comment (TheUnknownCod3r) [inline scripts/menu.gsc:540]: The Ray Gun and Mk2 Ray gun are Aether Only, so they need a check to ensure they only show up on Aether. This is Blood, Classified, Alpha, and Tag.

**Comment by TheUnknownCod3r** [comment]
Other than what I highlighted this is fine. If you can add a check for those maps, i'd be happy to merge it.

File: scripts/menu.gsc [hunk 2]
```diff
@@ -536,10 +536,39 @@ MenuOptionsPlayer(menu, player)
...
+            }
+            else if(BO4GetMap() == "AE"){
+                self addOpt("Hand of Ouranous", &GiveClientWeapon, "ww_hand_o_upgraded", self);
+                self addOpt("Hand of Hemera", &GiveClientWeapon, "ww_hand_h_upgraded", self);
+                self addOpt("Hand of Gaia", &GiveClientWeapon, "ww_hand_g_upgraded", self);
+                self addOpt("Hand of Charon", &GiveClientWeapon, "ww_hand_c_upgraded", self);
             }
+            else if(BO4GetMap() == "Tag"){
+                self addOpt("ThunderGun", &GiveClientWeapon, "thundergun_upgraded", self);
+                self addOpt("test", &GiveClientWeapon, "ww_tesla_gun_t8_upgraded", self);
```
Comment (TheUnknownCod3r) [inline scripts/menu.gsc:561]: Did this work? If so, you should get the PAP name, and use it for the option.

File: scripts/menu.gsc (hunk 1 above)
Comment (TubBuddy) [reply scripts/menu.gsc:542]: Done that now

File: scripts/menu.gsc (hunk 2 above)
Comment (TubBuddy) [reply scripts/menu.gsc:564]: Also done

**Comment by TubBuddy** [comment]
I could not fine the upgraded wunderwaffle sniper name 
Or any of the alistairs guns names

**Comment by TheUnknownCod3r** [comment]
> I could not fine the upgraded wunderwaffle sniper name Or any of the alistairs guns names

The Wonderwaffe sniper isn't what you used in the upgraded section, but in game, you can just call the Print Weapon Name function, and it should output the DisplayName to an iPrintLnBold for you in future. Ive corrected the Wonderwaffe name from memory, but it may be incorrect. Alistairs Folly and Alistairs Annihilator I believe are guns I have hashes for, not stringnames due to being unable to locate them in raw files, its just a guessing scenario to figure most of them out. I appreciate the work you've put in, thank you again.

**Comment by TheUnknownCod3r** [comment]
You can use this to pull the Weapon Display name, which will show you the name you see on screen, not the hash.

```
GetWeaponDisplayName()
{
    WeaponName = self GetCurrentWeapon().DisplayName;
    self iPrintLnBold(WeaponName);
}
```

**Comment by TubBuddy** [comment]
Ahh thanks! Also you forgot to remove upgraded from the AO rayguns btw haha

**Comment by TheUnknownCod3r** [comment]
good call, I missed that one. Fixed, and again, thanks :)

**Pull request merged**